File: fjt.api/models/Agreement.js
const Sequelize = require('sequelize');

module.exports = (sequelize) => {
    const Agreement = sequelize.define('Agreement', {
        agreementID: {
            allowNull: false,
            autoIncrement: true,
            primaryKey: true,
            type: Sequelize.INTEGER
        },
        agreementTypeID: {
            type: Sequelize.INTEGER,
            allowNull: false
        },
        agreementContent: {
            type: Sequelize.STRING,
            allowNull: false
        },
        version: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        system_variables: {
            type: Sequelize.STRING,
            allowNull: true
        },
        isPublished: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        publishedDate: {
            allowNull: true,
            defaultValue: null,
            type: Sequelize.DATE
        },
        isDeleted: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        createdBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        updatedBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        deletedBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        agreementSubject: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        createByRoleId: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        updateByRoleId: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        deleteByRoleId: {
            type: Sequelize.INTEGER,
            allowNull: true
        }
    },
        {
            tableName: 'agreement',
            paranoid: true
        });

    Agreement.associate = (models) => {
        Agreement.belongsTo(models.AgreementType, {
            foreignKey: 'agreementTypeID',
            as: 'agreementType'
        });
    };

    return Agreement;
};

File: FJT.Reporting/Models/Agreement.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FJT.Reporting.Models
{
    [Table("agreement")]
    public class Agreement
    {
        [Key]
        public int agreementID { get; set; }
        public int agreementTypeID { get; set; }
        public string agreementContent { get; set; }
        public int version { get; set; }
        public string system_variables { get; set; }
        public string agreementSubject { get; set; }
        public bool? isPublished { get; set; }
        public DateTime publishedDate { get; set; }
        public bool isDeleted { get; set; }
        [StringLength(255)]
        public string createdBy { get; set; }
        public DateTime createdAt { get; set; }
        [StringLength(255)]
        public string updatedBy { get; set; }
        public DateTime? updatedAt { get; set; }
        [StringLength(255)]
        public string deletedBy { get; set; }
        public DateTime? deletedAt { get; set; }
    }
}